scripts/build/gen_handles.py

```diff
@@ -72,35 +72,48 @@ def parse_args():
 
     sys.path.insert(0, os.path.join(ZEPHYR_BASE, "scripts/dts"))
 
+def sort_dev_handles(dev):
+    if len(dev.devs_depends_on) > 0:
+        dev.devs_depends_on_sorted = sorted(dev.devs_depends_on, key=lambda d: d.handle)
+
+    if len(dev.devs_depends_on_injected) > 0:
+        dev.devs_depends_on_injected_sorted = sorted(
+            dev.devs_depends_on_injected, key=lambda d: d.handle
+        )
+
+    if len(dev.devs_supports) > 0:
+        dev.devs_supports_sorted = sorted(dev.devs_supports, key=lambda d: d.handle)
+
+
 def c_handle_comment(dev):
     def dev_path_str(dev):
         return dev.edt_node and dev.edt_node.path or dev.sym.name
     lines = [
         '',
         '/* {:d} : {:s}:'.format(dev.handle, (dev_path_str(dev))),
     ]
-    if len(dev.devs_depends_on) > 0:
+    if len(dev.devs_depends_on_sorted) > 0:
         lines.append(' * Direct Dependencies:')
-        for dep in dev.devs_depends_on:
+        for dep in dev.devs_depends_on_sorted:
             lines.append(' *    - {:s}'.format(dev_path_str(dep)))
-    if len(dev.devs_depends_on_injected) > 0:
+    if len(dev.devs_depends_on_injected_sorted) > 0:
         lines.append(' * Injected Dependencies:')
-        for dep in dev.devs_depends_on_injected:
+        for dep in dev.devs_depends_on_injected_sorted:
             lines.append(' *    - {:s}'.format(dev_path_str(dep)))
-    if len(dev.devs_supports) > 0:
+    if len(dev.devs_supports_sorted) > 0:
         lines.append(' * Supported:')
-        for sup in dev.devs_supports:
+        for sup in dev.devs_supports_sorted:
             lines.append(' *    - {:s}'.format(dev_path_str(sup)))
     lines.append(' */')
     return lines
 
 def c_handle_array(dev, extra_support_handles=0):
```

Comment:
Couldn't all the changes in this PR be condensed down to:
```
handle_sort = lambda x: x.handle
# Handles are sorted to ensure the resulting C structures are reproducible.
handles = [
        *[str(d.handle) for d in sorted(dev.devs_depends_on, key=handle_sort)],
        ...
]
```
What I don't like about the current implementation is that is adds new member variables to the parser class, which aren't automatically populated, are only used for this script, and requires another function to be called to be valid.

Reply:
I started with that approach.  However, that results in a shuffle of device handles in the `__devicehdl[]` array compared to the preceding C comment.

So I can either duplicate the sort in both `c_handle_comment()` and `c_handle_array`.  Or I could change both `c_handle_comment()` and `c_handle_array` to accept 3 lists as arguments.

```
devs_depends_on_sorted = sorted(dev.devs_depends_on, key=lambda d: d.handle)
devs_depends_on_injected_sorted = sorted(dev.devs_depends_on_injected, key=lambda d: d.handle)
devs_supports_sorted = sorted(dev.devs_supports, key=lambda d: d.handle)

c_handle_comment(devs_depends_on_sorted, devs_depends_on_injected_sorted, devs_supports_sorted)
c_handle_array(devs_depends_on_sorted, devs_depends_on_injected_sorted, devs_supports_sorted)
```

Reply:
I agree it is probably a good idea for the comment order to be the same as the final array.
Your version is an improvement on the current implementation, I would probably write it as the following just to keep things grouped together:
```
handles = {
    "depends": sorted(dev.devs_depends_on, key=lambda d: d.handle),
    "injected": sorted(dev.devs_depends_on_injected, key=lambda d: d.handle),
    "supports": sorted(dev.devs_supports, key=lambda d: d.handle),
}
c_handle_comment(handles)
c_handle_array(handles)
```

Reply:
Thanks for the suggestion. This is much cleaner.